src/data/complex/basic.lean

```diff
@@ -524,4 +526,12 @@ lim_eq_of_equiv_const (λ ε ε0,
 let ⟨i, hi⟩ := equiv_lim f ε ε0 in
 ⟨i, λ j hj, lt_of_le_of_lt (abs_abs_sub_le_abs_sub _ _) (hi j hj)⟩)
 
+@[simp, norm_cast] lemma of_real_prod {α : Type*} (s : finset α) (f : α → ℝ) :
```

Comment:
Just out of curiosity, have you used these successfully as `norm_cast` lemmas? They're classified right and I don't see any reason in principle they shouldn't work. But I'm not sure the "casts under binders" case was explicitly considered in the `norm_cast` design.

Reply:
Yes, I have. See #4432. The implementation of `real.exp_bound` from `complex.exp_bound` using `norm_cast` relies on `of_real_sum` as a `norm_cast` lemma (as well as `abs_of_real` and `of_real_exp`; all the other `norm_cast` markers / lemmas in this PR were added by analogy rather than because I had an immediate use for them).